egs/avspeech/run.sh

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+
+set -e
+
+parse_boolean() {
+	# parses yaml boolean to flag.
+	# e.g. audio_set=True to --use-audio-set for argparse
+
+        var=$1
+        flag=$2
+        var=$(echo $var | awk '{print tolower($0)}')
+        var=$([ "$var" == "true" ] && echo "$flag" || echo "")
+        echo $var
+}
+
+get_attribute() {
+	# get an attribute from local/conf.yml
+
+	pushd $root_dir > /dev/null
+	attribute=$1
+	value=$(grep  $attribute local/data_prep.yml | awk  '{print $2}' | sed 's/"//g')
+	popd > /dev/null
+	echo $value
+}
+
+check_dependency() {
+	# check if dependeny exists for
+	# ffmpeg, sox, youtube-dl
+	cmd=$1
+
+	type $cmd  >/dev/null 2>&1 || { echo >&2 "$cmd is required but it's not installed. Aborting."; exit 1; }
+}
+
+root_dir=$(pwd)
+
+python_path=python
+loader_dir=$root_dir/local/loader
+stage=0
+gpu_ids=0
+exp_dir=exp/logdir
+tag=
+install_flag=false
+storage_dir=${STORAGE_DIR:-"storage_dir"}
+
+. utils/parse_options.sh
+
+if [ "$install_flag" = true ]; then
+	# install python dependencies
+	echo "installing python dependencies"
+	$python_path -m pip install -r local/requirements.txt > /dev/null
+fi
+
+if [[ $stage -le 0 ]]; then
+
+	echo "Stage 0: Setting up structure and downloading files."
+	pushd $loader_dir > /dev/null
+	# Setup the structure for data downloading, pre-processing
+	mkdir -p ../../$storage_dir/storage/{video,audio,embed,spec,mixed} ../../data/{audio_set,audio_visual} ../../exp ../../logs
+
+	n_jobs=$(get_attribute "download_jobs")
+	path=$(get_attribute "download_path")
+	download_start=$(get_attribute "download_start")
+	download_end=$(get_attribute "download_end")
+
+	echo "Downloading...(Interruptible)"
+	check_dependency ffmpeg
+	check_dependency youtube-dl
+	python3 download.py --jobs $n_jobs --path $path \
+			    --start $download_start \
+			    --end $download_end
+
+	n_files=$(ls -1q $vid_dir/*_final.mp4 | wc -l)
```

Comment:
vid_dir is not defined here..

Reply:
```suggestion
        vid_dir=$storage_dir/storage/video
	n_files=$(ls -1q $vid_dir/*_final.mp4 | wc -l)
```

Reply:
Yes, I did that as well but because of the constants path "problem", it couldn't go forward. But yes, that's perfect